src/test/test_pool_create.sh

```diff
@@ -34,15 +35,27 @@ function TEST_pool_create() {
     local poolname=rulepool
     local var=`ceph osd crush rule dump|grep -w ruleset|sed -n '$p'|grep -o '[0-9]\+'`
     var=`expr  $var + 1 `
-    ceph osd getcrushmap -o  map1
-    crushtool -d map1 -o map1.txt
+    ceph osd getcrushmap -o  $dir/map1
+    crushtool -d $dir/map1 -o $dir/map1.txt
 
     local minsize=0
     local maxsize=1
-    sed -i '/# end crush map/i\rule '$rulename' {\n ruleset \'$var'\n type replicated\n min_size \'$minsize'\n max_size \'$maxsize'\n step take default\n step choose firstn 0 type osd\n step emit\n }\n' map1.txt
-    crushtool  -c map1.txt -o  map1.bin
-    ceph osd setcrushmap -i map1.bin
-    ceph osd  pool create  $poolname 200 $rulename 2>rev
+	cat >> $dir/map1.txt << _EOF_RULE
+    rule $rulename {
+	    ruleset $var
+	    type replicated
+	    min_size $minsize
+	    max_size $maxsize
+	    step take default
+	    step choose firstn 0 type osd
+	    step emit
+	}
+_EOF_RULE
```

Comment:
@songbaisen 
Fixed the TABS, but also found another reference to a output file that was going to the current directory (rev).
Also prefixed that with $dir to get it into testdir.